src/Controller/Controller.php

```diff
@@ -390,6 +418,85 @@ public function __set($name, $value)
     }
 
     /**
+     * Returns the controller name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Sets the controller name.
+     *
+     * @param string $name Controller name.
+     * @return $this
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Returns the controller plugin name.
+     *
+     * @return string
+     */
+    public function getPlugin()
+    {
+        return $this->plugin;
+    }
+
+    /**
+     * Sets the controller plugin name.
+     *
+     * @param string $plugin Controller plugin name.
+     * @return $this
+     */
+    public function setPlugin($plugin)
+    {
+        $this->plugin = $plugin;
+
+        return $this;
+    }
+
+    /**
+     * Returns true if an action should be rendered automatically.
+     *
+     * @return bool
+     */
+    public function isAutoRenderEnabled()
+    {
+        return $this->autoRender;
+    }
+
+    /**
+     * Enable or disbale automatic action rendering.
+     *
+     * @param bool $autoRender Flag.
+     * @return $this
+     */
+    public function enableAutoRender($autoRender = true)
```

Comment:
If this method has `enable` prefix then it should only enable auto rendering. Currently doing `enableAutoRender(false)` would disable auto rendering which is not good for readability.